packages/seal/src/client.ts

```diff
@@ -340,12 +339,11 @@ export class SealClient {
 						continue;
 					}
 					this.#cachedKeys.set(`${fullId}:${server.objectId}`, keyElement);
-					receivedIds.add(fullId);
 				}
 
 				// Check if all the receivedIds are consistent with the requested fullIds.
 				// If so, consider the key server got all keys and mark as completed.
-				if (fullIds.every((fullId) => receivedIds.has(fullId))) {
+				if (fullIds.every((fullId) => this.#cachedKeys.has(`${fullId}:${server.objectId}`))) {
```

Comment:
This is unrelated to the CCA sec changes, but is to avoid the redundant `receivedIds` map.